js/main.js

```diff
@@ -0,0 +1,113 @@
+    NetworkTables.addRobotConnectionListener(onRobotConnection, true);
+
+
+    // sets a function that will be called when any NetworkTables key/value changes
+    NetworkTables.addGlobalListener(onValueChanged, true);
+
+    // hook up our SendableChoosers to combo boxes
+    attachSelectToSendableChooser("#auto-selector", "/SmartDashboard/Autonomous Mode");
+
+});
+
+function resetGyro() {
+
+    offsetGyro = currentGyro;
+    rotateCompass(currentGyro + Math.PI);
+}
+
+
+function updateLifterStatus(id) {
+    $("#l1").attr("class", "hidden-status");
+    $("#l2").attr("class", "hidden-status");
+    $("#l3").attr("class", "hidden-status");
+    $("#l4").attr("class", "hidden-status");
+    $("#" + id).attr("class", "inline");
+}
+
+function onValueChanged(key, value) {
+    switch (key) {
+        case "/SmartDashboard/gyro":
+            rotateCompass(value + Math.PI);
+            break;
+        case "/robot/mode":
+            if (value === "teleop") {
+                timerCycle();
+            }
+            if (value != "disabled") {
+                timerReset();
+            }
+            break;
+        case "/lifter/state":
+            if (value === "switch_height") {
+                updateLifterStatus("l1");
+            } else if (value === "lower_scale") {
+                updateLifterStatus("l2");
+            } else if (value === "balanced_scale") {
+                updateLifterStatus("l3");
+            } else if (value === "upper_scale") {
+                updateLifterStatus("l4");
+            }
+    }
+}
+
+function onRobotConnection(connected) {
+    if (connected) {
+        $("#connection").text("Connected");
+        $("#connection").css({
+            "color": "#4CAF50"
+        });
+    } else {
+        $("#connection").text("Disconnected");
+        $("#connection").css({
+            "color": "rgb(223, 48, 48)"
+        });
+
+    }
+}
+
+function remove_form() {
+    $(".checklist").hide();
+    $(".hidden").show();
+    $(".hiddengyro").show();
+}
+
+function timerCycle() {
+    var countDownDate = Math.floor(Date.now() / 1000) + 135;
+    setInterval(function() {
+        var now = Math.floor(Date.now() / 1000);
+        var difference = countDownDate - now;
+
+        if (difference <= 0) {
+            document.getElementById("timer").innerHTML = "";
+            $("#timer").text("GOOD JOB!");
+            $("#timer").css("font-size", "425%");
+            $("#timer").css("color", "#4CAF50");
+            $("#timer").toggleClass("blink");
+        } else if (difference < 10) {
+            document.getElementById("timer").innerHTML = "00" + difference;
+        } else if (difference < 100) {
+            document.getElementById("timer").innerHTML = "0" + difference;
+        } else {
+            document.getElementById("timer").innerHTML = difference;
+        }
+    }, 1000);
+}
+function timerReset(){
+    $('#timer').text('135');
+}
```

Comment:
I don't think this timer actually behaves correctly. I don't imagine it'll be very useful for the drivers either (especially if it doesn't behave correctly).

Reply:
Yes this will only replace the timer to say 135 for one second then will continue counting down from whatever it was when the reset button was pressed